sycl/source/detail/sycl_mem_obj_t.cpp

```diff
@@ -90,8 +90,12 @@ void SYCLMemObjT::updateHostMemory() {
 
   // If we're attached to a memory record, process the deletion of the memory
   // record. We may get detached before we do this.
-  if (MRecord)
-    assert(Scheduler::getInstance().removeMemoryObject(this));
+  if (MRecord) {
+    bool Result = Scheduler::getInstance().removeMemoryObject(this);
+    assert(
+        Result &&
+        "removeMemoryObject should not return false in mem object destructor");
+  }
```

Comment:
There will be a warning saying that `Result` is unused. And will turn into an error when building with `-werror`.